drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua

```diff
@@ -164,130 +164,100 @@ local supported_profiles =
+
   -- If all endpoints are supported, use '-all' in the profile name so that it
   -- remains under the profile name character limit
-  if level_indication_support == "-co-co2-no2-ozone-formaldehyde-pm1-pm25-pm10-radon-tvoc" then
-    level_indication_support = "-all"
+  if level_name == "-co-co2-no2-ozone-ch2o-pm1-pm25-pm10-radon-tvoc" then
+    level_name = "-all"
   end
-  if level_indication_support ~= "" then
-    profile_name = profile_name .. level_indication_support .. "-level"
+  if level_name ~= "" then
+    profile_name = profile_name .. level_name .. "-level"
   end
 
-  if #co_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-co"
-  end
-  if #co2_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-co2"
-  end
-  if #no2_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-no2"
-  end
-  if #ozone_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-ozone"
-  end
-  if #formaldehyde_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-formaldehyde"
-  end
-  if #pm1_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-pm1"
-  end
-  if #pm2_5_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-pm25"
-  end
-  if #pm10_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-pm10"
-  end
-  if #radon_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-radon"
-  end
-  if #tvoc_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-tvoc"
-  end
   -- If all endpoints are supported, use '-all' in the profile name so that it
   -- remains under the profile name character limit
-  if numeric_measurement_support == "-co-co2-no2-ozone-formaldehyde-pm1-pm25-pm10-radon-tvoc" then
-    numeric_measurement_support = "-all"
+  if meas_name == "-co-co2-no2-ozone-ch2o-pm1-pm25-pm10-radon-tvoc" then
+    meas_name = "-all"
   end
-  if numeric_measurement_support ~= "" then
-    profile_name = profile_name .. numeric_measurement_support .. "-meas"
+  if meas_name ~= "" then
+    profile_name = profile_name .. meas_name .. "-meas"
   end
 
   if not tbl_contains(supported_profiles, profile_name) then
     device.log.warn_with({hub_logs=true}, string.format("No matching profile for device. Tried to use profile %s", profile_name))
-    if #co_meas_eps > 0 or #no2_meas_eps > 0 or #ozone_meas_eps > 0 or #formaldehyde_meas_eps > 0 or
-        #pm1_meas_eps > 0 or #pm10_meas_eps > 0 or #radon_meas_eps > 0 then
+
+    local function meas_find(sub_name)
+      return string.match(meas_name, sub_name)
+    end
+
+    -- try to best match to existing profiles
+    if profile_name == "aqs-co" or meas_find("co-") or meas_find("no2") or meas_find("ozone") or meas_find("ch2o") or
```

Comment:
Should this just be `meas_find("co")` instead of `meas_find("co-")`?

```suggestion
    if profile_name == "aqs-co" or meas_find("co") or meas_find("no2") or meas_find("ozone") or meas_find("ch2o") or
```

Reply:
No, but I can leave a comment to explain that decision. If it were just `meas_find("co")` that would catch `co2` as well as `co`, which we don't want if we want to copy the current logic,

Reply:
Also, could you help me understand how the profile name `aqs-co` could be constructed from above? I'm not sure I understand why that profile is called out specifically in this conditional

Reply:
The one case where `co` wouldn't be caught with a - following it is when it's the only addition,aka`aqs-co`, which is why that is also checked.

Reply:
Yes! My second comment explained that, but I definitely think this should include a comment to describe this rare case

Reply:
@ctowns I have updated the explanation per our conversation, and I've edited the logic a little. I removed this line `profile_name == "aqs-co"` in place of `meas_name:match("co$")`, since the original conditional was actually a little wrong :sweat_smile: since it would've also needed to check the case swhere `profile_name` included the permutations `-humidity` as well as `-temp`.